src/creature.cpp

```diff
@@ -712,11 +716,11 @@ BlockType_t Creature::blockHit(Creature* attacker, CombatType_t combatType, int3
 		}
 
 		if (combatType != COMBAT_HEALING) {
-			attacker->onAttackedCreature(this);
+			attacker->onAttackedCreature(getCreature());
 			attacker->onAttackedCreatureBlockHit(blockType);
 			if (attacker->getMaster() && attacker->getMaster()->getPlayer()) {
-				Player* masterPlayer = attacker->getMaster()->getPlayer();
-				masterPlayer->onAttackedCreature(this);
+				auto masterPlayer = attacker->getMaster()->getPlayer();
```

Comment:
Could this be `const auto`
```suggestion
				const auto masterPlayer = attacker->getMaster()->getPlayer();
```